virtualenv.py

```diff
@@ -900,98 +924,105 @@ def find_wheels(projects, search_dirs):
 
 def install_wheel(project_names, py_executable, search_dirs=None, download=False):
     if search_dirs is None:
-        search_dirs = file_search_dirs()
+        search_dirs_context = virtualenv_support_dirs
+    else:
 
-    wheels = find_wheels(["setuptools", "pip"], search_dirs)
-    python_path = os.pathsep.join(wheels)
+        @contextlib.contextmanager
+        def search_dirs_context():
+            yield search_dirs
 
-    # PIP_FIND_LINKS uses space as the path separator and thus cannot have paths
-    # with spaces in them. Convert any of those to local file:// URL form.
-    try:
-        from urlparse import urljoin
-        from urllib import pathname2url
-    except ImportError:
-        from urllib.parse import urljoin
-        from urllib.request import pathname2url
+    with search_dirs_context() as search_dirs:
+        wheels = find_wheels(["setuptools", "pip"], search_dirs)
```

Comment:
this generates a long diff because of the indent, what if we would just extract the indented part into its own function?

Reply:
This is why I suggested viewing the diff with [`?w=1`](https://github.com/pypa/virtualenv/pull/1287/files?w=1) in [this comment](https://github.com/pypa/virtualenv/pull/1287/files?w=1)

Reply:
Not if you move it to a private function, right after this function. The private function call would be an insert and everything below would be all is same.

Reply:
I'd rather not add lines for no reason, what's the problem with the indent?